syntax = "proto3";

package kvstore;

service KeyValueService {
  rpc Put(PutRequest) returns (PutResponse) {}
  rpc Get(GetRequest) returns (GetResponse) {}
}

message PutRequest {
  string key = 1;
  bytes value = 2;
}

message PutResponse {
  bool success = 1;
  string message = 2;
}

message GetRequest { string key = 1; }

message GetResponse {
  bytes value = 1;
  bool found = 2;
}
